spectral_cube/spectral_cube.py

```diff
@@ -237,11 +241,28 @@ def __init__(self, data, wcs, mask=None, meta=None, fill_value=np.nan,
         # Deal with metadata first because it can affect data reading
         self._meta = meta or {}
         if 'BUNIT' in self._meta:
-            try:
-                self._unit = u.Unit(self._meta['BUNIT'])
-            except ValueError:
-                warnings.warn("Could not parse unit {0}".format(self._meta['BUNIT']))
-                self._unit = None
+
+            # special case: CASA makes non-FITS-compliant jy/beam headers
```

Comment:
it may now be best to split out the metadata parsing into a separate function or even separate .py file